server/lib/util.js

```diff
@@ -118,3 +119,15 @@ function dateFormatter(rows, dateFormat) {
   return rows;
 }
 
+//check if an object has no key
+function isEmpty(map) {
```

Comment:
There's a lodash function for this [_.empty()](https://lodash.com/docs/4.17.4#isEmpty).  No need to re-implement it.